### DISCUSSION TITLE
Can pulsar provide a broker plug-in to rewrite the payload?

### DISCUSSION THREAD

**Discussion by chenyahui** [apache/pulsar#21827] [Q&A] [answered]
Now Pulsar has a broker plug-in named EntryFilter, which allows users to customize filtering logic and filter data that consumers do not need on the broker side.

But this is not completely sufficient: in many scenarios, we also need to streamline the payload on the broker side. The broker directly cuts out the parts of the payload that the Consumer does not need.

For example: Payload is `{"a": 1, "b": 2, "c": 3}`, and Consumer only needs `{"a": 1}`. We can directly cut the data to `{"a": 1}` on the broker side. {"a": 1}`, and then send it to Consumer, This can reduce a lot of IO bandwidth.

Therefore, can pulsar provide a broker plug-in to rewrite the payload?

**Comment by dao-jun** [comment, marked as answer]
It doesn't make sense. 
If broker provides such a plugin, which means broker will decode the payload, it's not recommended.
It will lead to broker CPU and Memory usage increasing.
